src/ModelContextProtocol/McpJsonUtilities.cs

```diff
@@ -78,7 +78,11 @@ internal static bool IsValidMcpToolSchema(JsonElement element)
     // Keep in sync with CreateDefaultOptions above.
     [JsonSourceGenerationOptions(JsonSerializerDefaults.Web,
         DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
-        NumberHandling = JsonNumberHandling.AllowReadingFromString)]
+        NumberHandling = JsonNumberHandling.AllowReadingFromString
+#if NET9_0_OR_GREATER
+        ,UseStringEnumConverter = true
```

Comment:
I don't think we want different behaviors here based on which build is being used.
cc: @eiriktsarpalis